IMPORTING LIBRARYS
=

In [109]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [110]:
df = pd.read_csv('zomato.csv')
df.head(2)
print('4SF20CI021')

4SF20CI021


In [111]:
df.columns

Index(['url', 'address', 'name', 'online_order', 'book_table', 'rate', 'votes',
       'phone', 'location', 'rest_type', 'dish_liked', 'cuisines',
       'approx_cost(for two people)', 'reviews_list', 'menu_item',
       'listed_in(type)', 'listed_in(city)'],
      dtype='object')

DROPPING SOME UNWANTED COLUMNS
=

In [112]:
df = df.drop(['url', 'address', 'phone', 'menu_item', 'dish_liked', 'reviews_list','listed_in(city)'], axis = 1)

In [113]:
df.head(2)

name online_order book_table   rate  votes      location  \
0           Jalsa          Yes        Yes  4.1/5    775  Banashankari   
1  Spice Elephant          Yes         No  4.1/5    787  Banashankari   

       rest_type                        cuisines approx_cost(for two people)  \
0  Casual Dining  North Indian, Mughlai, Chinese                         800   
1  Casual Dining     Chinese, North Indian, Thai                         800   

  listed_in(type)  
0          Buffet  
1          Buffet

In [114]:
df.info()
print('4SF20CI021')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51717 entries, 0 to 51716
Data columns (total 10 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   name                         51717 non-null  object
 1   online_order                 51717 non-null  object
 2   book_table                   51717 non-null  object
 3   rate                         43942 non-null  object
 4   votes                        51717 non-null  int64 
 5   location                     51696 non-null  object
 6   rest_type                    51490 non-null  object
 7   cuisines                     51672 non-null  object
 8   approx_cost(for two people)  51371 non-null  object
 9   listed_in(type)              51717 non-null  object
dtypes: int64(1), object(9)
memory usage: 3.9+ MB
4SF20CI021


In [115]:
df.shape

(51717, 10)

In [116]:
df.drop_duplicates(inplace = True)
df.dropna(inplace = True)
df.shape

(30958, 10)

WE CAN OBSERVE THAT THE NUMBER OF ROWS IS REDUCED THAT MEANS THERE WERE DUPLICATE ROWS HENCE THEY ARE DROPPED(51717 --> 51609)
=
NOW LETS US CLEAN THE EACH COLUMN  IN THE DATSET STARTING WITH RATE COLUMN
=

In [117]:
df['rate'].unique()
print('4SF20CI021')

4SF20CI021


WE CAN OBSERVE THAT THE VALUES IS IN FLOAT , BUT SOME ARE NOT HENCE WE WRITE A FUNCTION handlerate() TO CONVERT IT INTO FLOAT VALUES. ALSO, SET THE RATINGS "NEW" AS MEAN.
=

In [118]:
def handlerate(value):
    if(value=='NEW' or value=='-'):
        return np.nan
    else:
        value = str(value).split('/')
        value = value[0]
        return float(value)
    
df['rate'] = df['rate'].apply(handlerate)
print('4SF20CI021')

4SF20CI021


In [119]:
df['rate'].fillna(df['rate'].mean(), inplace = True)
df['rate'].unique()

array([4.1       , 3.8       , 3.7       , 3.6       , 4.6       ,
       4.        , 4.2       , 3.9       , 3.1       , 3.        ,
       3.2       , 3.3       , 2.8       , 4.4       , 4.3       ,
       3.73284972, 2.9       , 3.5       , 2.6       , 3.4       ,
       4.5       , 2.5       , 2.7       , 4.7       , 2.4       ,
       2.2       , 2.3       , 4.8       , 4.9       , 2.1       ,
       2.        , 1.8       ])

In [120]:
df.rename(columns = {'approx_cost(for two people)':'Cost2plates', 'listed_in(type)':'Type'}, inplace = True)

LETS CLEAN THE NEXT COLUMN 'LOCATION' SINCE EVERYHTING LOOKS FINE...LETS MOVE TO THE NEXT COLUMN 'Cost2plates'
=

In [121]:
df['location'].unique()
print('4SF20CI021')

4SF20CI021


In [122]:
df['Cost2plates'].unique()

array(['800', '300', '600', '700', '550', '500', '450', '650', '400',
       '900', '200', '750', '150', '850', '100', '1,200', '350', '250',
       '950', '1,000', '1,500', '1,300', '199', '80', '1,100', '160',
       '1,600', '230', '130', '1,700', '1,400', '1,350', '2,200', '2,000',
       '1,800', '1,900', '180', '330', '2,500', '2,100', '3,000', '2,800',
       '3,400', '50', '40', '1,250', '3,500', '4,000', '2,400', '2,600',
       '1,450', '70', '3,200', '560', '240', '360', '6,000', '1,050',
       '2,300', '4,100', '120', '5,000', '3,700', '1,650', '2,700',
       '4,500'], dtype=object)

SINCE HAVING A ',' IN NUMBER MORE THAN 3 DIGITS WILL CAUSE ERROR IN PLOTTING WE REMOVE IT  USING handlecomma() FUNCTION.
=

In [123]:
def handlecomma(value):
    value = str(value)
    if ',' in value:
        value = value.replace(',', '')
        return float(value)
    else:
        return float(value)
    
df['Cost2plates'] = df['Cost2plates'].apply(handlecomma)
df['Cost2plates'].unique()
print('4SF20CI021')

4SF20CI021


MOVING TO THE NEXT COLUMN i.e,'rest_type'.HERE WE CAN FIND THAT THAT VALUES RANGE A LOT ie, FROM 1 TO 1000+.HENCE LETS MAKE THAT COUNT AS 'OTHERS' WHOSE VALUE IS LESS THAN 1000 USING handle_rest_type
=

In [124]:
rest_types = df['rest_type'].value_counts(ascending  = False)
rest_types

Quick Bites                   10143
Casual Dining                  7311
Cafe                           2495
Dessert Parlor                 1357
Delivery                       1117
                              ...  
Dessert Parlor, Kiosk             2
Bakery, Food Court                2
Food Court, Beverage Shop         2
Quick Bites, Kiosk                1
Dessert Parlor, Food Court        1
Name: rest_type, Length: 87, dtype: int64

In [125]:
rest_types_lessthan1000 = rest_types[rest_types<1000]
rest_types_lessthan1000
print('4SF20CI021')

4SF20CI021


In [126]:
def handle_rest_type(value):
    if(value in rest_types_lessthan1000):
        return 'others'
    else:
        return value
        
df['rest_type'] = df['rest_type'].apply(handle_rest_type)
df['rest_type'].value_counts()

Quick Bites       10143
others             8535
Casual Dining      7311
Cafe               2495
Dessert Parlor     1357
Delivery           1117
Name: rest_type, dtype: int64

SIMILARLY,WITH 'LOCATION' COLUMN VALES LESS THAN 300 IS CHANGED TO 'OTHERS'. AND IN 'CUISINE' COLUMN IF IT IS LESS THAN 100
=

In [127]:
location = df['location'].value_counts(ascending  = False)

location_lessthan300 = location[location<300]



def handle_location(value):
    if(value in location_lessthan300):
        return 'others'
    else:
        return value
        
df['location'] = df['location'].apply(handle_location)
df['location'].value_counts()
print('4SF20CI021')

4SF20CI021


In [128]:
cuisines = df['cuisines'].value_counts(ascending  = False)


cuisines_lessthan100 = cuisines[cuisines<100]



def handle_cuisines(value):
    if(value in cuisines_lessthan100):
        return 'others'
    else:
        return value
        
df['cuisines'] = df['cuisines'].apply(handle_cuisines)
df['cuisines'].value_counts()
print('4SF20CI021')

4SF20CI021


DATA IS CLEAN NOW...HENCE NOW WE CAN USE IT FOR FURTHER  ANALYSIS.
=